worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/BlockArrayClipboard.java

```diff
@@ -197,13 +198,49 @@ public boolean setBiome(int x, int y, int z, BiomeType biome) {
     public List<? extends Entity> getEntities(Region region) {
         region = region.clone();
         region.shift(BlockVector3.ZERO.subtract(origin));
-        return getParent().getEntities(region);
+        return getParent().getEntities(region).stream().map(e ->
+        {
+            if (e instanceof ClipboardEntity) {
+                ClipboardEntity ce = (ClipboardEntity) e;
+                Location oldloc = ce.getLocation();
+                Location loc = new Location(oldloc.getExtent(),
+                                            oldloc.getX() + origin.getBlockX(),
+                                            oldloc.getY() + origin.getBlockY(),
+                                            oldloc.getZ() + origin.getBlockZ(),
+                                            oldloc.getYaw(), oldloc.getPitch());
+                return new ClipboardEntity(loc, ce.entity);
+            }
+            return e;
+        }).collect(Collectors.toList());
     }
 
+    @Override
+    public List<? extends Entity> getEntities() {
+        return getParent().getEntities().stream().map(e ->
+        {
```

Comment:
ew

Reply:
the block positions are shifted as well, see getBlock, getFullBlock, etc